frontend/azlinux/handle_bin.go

```diff
@@ -0,0 +1,121 @@
+package azlinux
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/dalec"
+	"github.com/Azure/dalec/frontend"
+	"github.com/Azure/dalec/frontend/rpm"
+	"github.com/moby/buildkit/client/llb"
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+	ocispecs "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+func binCopyScript(rpms []string, binaries map[string]dalec.ArtifactConfig) string {
+	var sb strings.Builder
+
+	sb.WriteString(`
+#!/bin/bash
+set -e
+declare -a RPMS=()
+export RPM_BINDIR=$(rpm --eval '%{_bindir}')
+`)
+
+	for _, rpm := range rpms {
+		sb.WriteString(fmt.Sprintf("RPMS+=(%q)\n", rpm))
+	}
+
+	sb.WriteString("for rpm in $RPMS; do\n")
+	binaryPathList := make([]string, 0, len(binaries))
+	for path, bin := range binaries {
+		srcPath := bin.InstallPath(path)
+		binaryPathList = append(binaryPathList, filepath.Join(".$RPM_BINDIR", srcPath))
+	}
+
+	sb.WriteString(fmt.Sprintf("rpm2cpio /package/RPMS/$rpm | cpio -imvd -D /extracted %s\n",
+		strings.Join(binaryPathList, " ")))
+	sb.WriteString("done\n")
+
+	sb.WriteString(
+		strings.Join([]string{
+			`export FILES=$(find ./extracted -type f)`,
+			`[[ -z $FILES ]] && (echo 'No binaries found to extract'; exit 1)`,
+			`cp ${FILES} /out`,
+		}, "\n"),
+	)
+	sb.WriteByte('\n')
+
+	return sb.String()
+}
+
+func zip(worker llb.State, zipName string, outputDir string, artifacts llb.State) llb.State {
+	outName := filepath.Join(outputDir, zipName+".zip")
+	return worker.Run(
+		shArgs("zip "+outName+" *"),
+		llb.Dir("/tmp/artifacts"),
+		llb.AddMount("/tmp/artifacts", artifacts)).AddMount(outputDir, llb.Scratch())
+}
```

Comment:
Can't this `zip` function be deleted and use directly the `Zip` from `dalec` package in the `helpers.go` file? These seem to be the same, no?

Reply:
Ah yes thank you, I think I meant to delete this